app/src/main/java/me/tangobee/weathernaut/MainActivity.kt

```diff
@@ -27,38 +31,54 @@ import kotlin.system.exitProcess
 class MainActivity : AppCompatActivity() {
 
     private lateinit var binding: ActivityMainBinding
-
     private lateinit var weatherViewModel: WeatherViewModel
     private lateinit var coroutineExceptionHandler: CoroutineExceptionHandler
     private lateinit var sharedPreferencesHelper: SharedPreferencesHelper
 
     private var settingsUpdated = false
 
+    // For requesting notification permissions
+    private val requestPermissionLauncher =
+        registerForActivityResult(ActivityResultContracts.RequestPermission()) { isGranted: Boolean ->
+            if (isGranted) {
+                Toast.makeText(this, "Notification permission granted", Toast.LENGTH_SHORT).show()
+            } else {
+                Toast.makeText(this, "Notification permission denied", Toast.LENGTH_SHORT).show()
+            }
+        }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         val splashScreen = installSplashScreen()
         binding = ActivityMainBinding.inflate(layoutInflater)
 
         sharedPreferencesHelper = SharedPreferencesHelper(this)
 
-        val noInternetLiveData : MutableLiveData<Boolean> =  MutableLiveData(false)
+        val noInternetLiveData : MutableLiveData<Boolean> = MutableLiveData(false)
 
-        coroutineExceptionHandler = CoroutineExceptionHandler{_, throwable ->
-            if(throwable is UnknownHostException) {
+        // Handle internet-related errors
+        coroutineExceptionHandler = CoroutineExceptionHandler { _, throwable ->
+            if (throwable is UnknownHostException) {
                 noInternetLiveData.postValue(true)
             }
         }
 
-        noInternetLiveData.observe(this) {noInternet ->
-            if(noInternet) {
-                Toast.makeText(this, getString(R.string.no_internet), Toast.LENGTH_LONG).show()
-                Thread.sleep(1000)
-                exitProcess(0)
+        noInternetLiveData.observe(this) { noInternet ->
+            if (noInternet) {
+                Snackbar.make(binding.root, getString(R.string.no_internet), Snackbar.LENGTH_INDEFINITE)
+                    .setAction(getString(R.string.retry)) {
+                        fetchData() // Retry fetching data
+                    }.show()
```

Comment:
Can you remove that Snackbar? When I am trying to open my app without internet I am not able to see any Snackbar.

Reply:
i have removed that snackbar tangobeee now the notification should work to stop music plsu the toast messages without any unnecessary snackbar code